File: viajerosrepojava-master/viajerosrepojava-master/viajeros/viajeros/src/main/java/ps/com/viajeros/dtos/viaje/ViajesRequestMatchDto.java
package ps.com.viajeros.dtos.viaje;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ViajesRequestMatchDto {
    private String origin;
    private String destination;
    private Boolean petsAllowed;
    private Boolean smokersAllowed;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime fechaHoraInicio;
    private Long equipajePermitido;
    private Long localidadInicioId;
    private Long localidadFinId;
    // Getters y setters
}

File: viajerosrepoangular-master/viajerosrepoangular-master/viajeros/src/app/models/Viajes/ViajesRequestMatchDto.ts
export class ViajesRequestMatchDto {
    origin: number;
    destination: number;
    petsAllowed: boolean;
    smokersAllowed: boolean;
    fechaHoraInicio: Date;
    equipajePermitido: number;
    localidadInicioId: number;
    localidadFinId: number;
  
    constructor(
      origin: number,
      destination: number,
      petsAllowed: boolean,
      smokersAllowed: boolean,
      fechaHoraInicio: Date,
      equipajePermitido: number,
      localidadInicioId: number,
      localidadFinId: number
    ) {
      this.origin = origin;
      this.destination = destination;
      this.petsAllowed = petsAllowed;
      this.smokersAllowed = smokersAllowed;
      this.fechaHoraInicio = fechaHoraInicio;
      this.equipajePermitido = equipajePermitido;
      this.localidadInicioId = localidadInicioId;
      this.localidadFinId = localidadFinId;
    }
  }
  